src/dataconnect/errors.ts

```diff
@@ -0,0 +1,42 @@
+import { GraphqlError, IncompatibleSqlSchemaError } from "./types";
+
+const INCOMPATIBLE_SCHEMA_ERROR_TYPESTRING =
+  "type.googleapis.com/google.firebase.dataconnect.v1main.IncompatibleSqlSchemaError";
+
+const GRAPHQL_ERROR_TYPESTRING =
+  "type.googleapis.com/google.firebase.dataconnect.v1main.GraphqlError";
+
+export function getIncompatibleSchemaError(err: any): IncompatibleSqlSchemaError | undefined {
+  const original = err.context?.body?.error || err.orignal;
+  if (!original) {
+    // If we can't get the original, rethrow so we don't cover up the original error.
+    throw err;
+  }
+  const details: any[] = original.details;
+  const incompatibles = details.filter((d) => d["@type"] === INCOMPATIBLE_SCHEMA_ERROR_TYPESTRING);
+  // Should never get multiple incompatible schema errors
+  return incompatibles[0];
+}
+
+export function isInvalidConnectorError(err: { message?: string }): boolean {
+  return err.message?.includes("The request was invalid: invalid connector") ?? false;
```

Comment:
Relying on the error message is pretty bad.

LG for now. Though let me try to make backend return proper failed precondition error details and code "INVALID_CONNECTOR".

This change would probably make the cut off.

http://shortn/_9PGdmqCjVx

Reply:
Now using these new errors.